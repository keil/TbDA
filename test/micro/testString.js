// String objects
// 15.5.1.1
var a = String();
var b = String(true);
var c = String(77);

dumpValue(a);
dumpValue(b);
dumpValue(c);
dumpValue(c.valueOf());

var d = new String();
var e = new String(false);
var f = new String(77.77);

dumpValue(d);
dumpValue(d.valueOf());
dumpValue(d.length);
dumpValue(e);
dumpValue(e.valueOf());
dumpValue(e.toUpperCase());
dumpValue(e.length);
dumpValue(f);
dumpValue(f.toString());
dumpValue(f.length);

dumpValue(String.fromCharCode());
dumpValue(String.fromCharCode(65, 66, 67, 68, 69, 70));

dumpValue(f.charAt(2));
dumpValue(f.charAt(140));

dumpValue(f.charCodeAt(2));
dumpValue(f.charCodeAt(140));
var g = new Number(42.125);
g.charCodeAt = f.charCodeAt;
dumpValue(g.charCodeAt(3));


dumpValue(f.concat(e, "finish"));
g.concat = f.concat;
dumpValue(g.concat(g));

var h = new String("testing indexOf");
dumpValue(h.indexOf("i"));
dumpValue(h.indexOf("i", 7));
dumpValue(h.indexOf("i", 40));
g.indexOf = h.indexOf;
dumpValue(g.indexOf("."));

dumpValue(h.lastIndexOf("i"));
dumpValue(h.lastIndexOf("i", 7));
dumpValue(h.lastIndexOf("i", 1));
g.lastIndexOf = h.lastIndexOf;
dumpValue(g.lastIndexOf("."));

var i = new String("testing localeCompare");
dumpValue(i.localeCompare(i));
dumpValue(i.localeCompare("test"));
dumpValue(i.localeCompare("utest"));
dumpValue(i.localeCompare());

// var j = new String("testing match");
// dumpValue(j.match(/test/));
// dumpValue(j.match(new RegExp("test")));
