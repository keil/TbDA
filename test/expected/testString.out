test/micro/testString.js:61: [definite] Too few parameters to native function String.prototype.localeCompare
test/micro/testString.js:10: [definite] Converting primitive string to object
test/micro/testString.js:10: [definite] Converting primitive string to object
test/micro/testString.js:4: [maybe] Converting boolean to string
test/micro/testString.js:13: [maybe] Converting boolean to string
test/micro/testString.js:7: [info] Abstract value: ""
test/micro/testString.js:8: [info] Abstract value: "true"
test/micro/testString.js:9: [info] Abstract value: "77"
test/micro/testString.js:10: [info] Abstract value: "77"
test/micro/testString.js:16: [info] Abstract value: [@String#node29]
test/micro/testString.js:17: [info] Abstract value: ""
test/micro/testString.js:18: [info] Abstract value: 0.0
test/micro/testString.js:19: [info] Abstract value: [@String#node33]
test/micro/testString.js:20: [info] Abstract value: "false"
test/micro/testString.js:21: [info] Abstract value: "FALSE"
test/micro/testString.js:22: [info] Abstract value: 5.0
test/micro/testString.js:23: [info] Abstract value: [@String#node37]
test/micro/testString.js:24: [info] Abstract value: "77.77"
test/micro/testString.js:25: [info] Abstract value: 5.0
test/micro/testString.js:27: [info] Abstract value: ""
test/micro/testString.js:28: [info] Abstract value: "ABCDEF"
test/micro/testString.js:30: [info] Abstract value: "."
test/micro/testString.js:31: [info] Abstract value: ""
test/micro/testString.js:33: [info] Abstract value: 46.0
test/micro/testString.js:34: [info] Abstract value: NaN
test/micro/testString.js:37: [info] Abstract value: 49.0
test/micro/testString.js:40: [info] Abstract value: Str
test/micro/testString.js:42: [info] Abstract value: Str
test/micro/testString.js:45: [info] Abstract value: UInt|NotUInt
test/micro/testString.js:46: [info] Abstract value: UInt|NotUInt
test/micro/testString.js:47: [info] Abstract value: UInt|NotUInt
test/micro/testString.js:49: [info] Abstract value: UInt|NotUInt
test/micro/testString.js:51: [info] Abstract value: UInt|NotUInt
test/micro/testString.js:52: [info] Abstract value: UInt|NotUInt
test/micro/testString.js:53: [info] Abstract value: UInt|NotUInt
test/micro/testString.js:55: [info] Abstract value: UInt|NotUInt
test/micro/testString.js:58: [info] Abstract value: Num
test/micro/testString.js:59: [info] Abstract value: Num
test/micro/testString.js:60: [info] Abstract value: Num
test/micro/testString.js:61: [info] Abstract value: Num
